src/serviceWorker.ts

```diff
@@ -0,0 +1,25 @@
+// Needed per Vite PWA docs
+import { precacheAndRoute } from 'workbox-precaching'
+declare let self: ServiceWorkerGlobalScope
+precacheAndRoute(self.__WB_MANIFEST)
+
+// Receive push notifications
+self.addEventListener('push', function (event) {
+    
+    if (!(
+        self.Notification &&
+        self.Notification.permission === 'granted'
+    )) {
+        //notifications aren't supported or permission not granted!
+        console.log("Notifications aren't supported or permission not granted!");
+        return;
+    }
+
+    if (event.data) {
+        console.log(event);
+        let message = event.data.json();
+        self.registration.showNotification(message.title, {
+            body: message.body,
```

Comment:
there are a bunch of options on this which we might want to configure, such as `timestamp`.

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification

There are also other parts for which having setting might be nice.
We can make this another PR if you'd like

Reply:
prefer another PR. we can add url that opens up the monitor aswell. will be fun to look into